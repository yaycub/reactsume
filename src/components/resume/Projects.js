import React from 'react';
import PropTypes from 'prop-types';
import ProjectItem from './ProjectItem';

const Projects = ({ projects }) => {
  const projectsElement = projects.map((project, i) => {
    return (
      <li key={i}>
        <ProjectItem {...project} />
      </li>
    );
  });

  return (
    <ul>
      {projectsElement}
    </ul>
  );
};

Projects.propTypes = {
  projects: PropTypes.arrayOf(PropTypes.shape({
    title: PropTypes.string.isRequired,
    tech: PropTypes.string,
    description: PropTypes.string.isRequired
  })).isRequired
};

export default Projects;
