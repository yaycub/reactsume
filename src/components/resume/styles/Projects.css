.Projects{
  display: flex;
  flex-direction: column;

  h3{
    font-size: 15px;
    font-weight: bold;
    margin-bottom: 10px;
    border-bottom: 1px dotted black;
  }

  h4{
    font-size: 13px;
    margin-bottom: 5px;
    font-weight: bold;
  }

  li{
    font-size: 12px;
    margin-top: 2px;
    margin-bottom: 2px;
  }

  p{
    margin-bottom: 5px;
  }
}