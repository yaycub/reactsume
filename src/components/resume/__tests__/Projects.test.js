import React from 'react';
import { shallow } from 'enzyme';
import data from '../../../resume.json';
import Projects from '../Projects.js';

describe('Projects Component', () => {
  it('can render Projects', () => {
    const wrapper = shallow(<Projects {...data} />);
    expect(wrapper).toMatchSnapshot();
  });
});
