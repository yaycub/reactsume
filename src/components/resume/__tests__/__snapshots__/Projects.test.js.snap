// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Projects Component can render Projects 1`] = `
<ul
  className="Projects"
>
  <h3>
    Projects
  </h3>
  <li
    key="0"
  >
    <ProjectItem
      description="Used a Markov Chain to creative a predictive text Twitter bot that used the material of numerous notable authors found in the Creative Commons. This was a group effort in which we mob programmed a majority of the app. My primarily contributions was working on the Mongoose Models/Methods and Twit NPM."
      tech="Node.js, Express, MongoDB, Mongoose, Jest, Markov"
      title="Markov Twain"
    />
  </li>
  <li
    key="1"
  >
    <ProjectItem
      description="This was a group project with the idea of making news more palatable. The news was pulled from the News API and presented to the user in a card stack that was swipeable. This involved user authentication and would store users saved articles within a Postgres database.  My primary contribution was creating the Node.js server using Express CRUD routes and a PostgreSQL Database."
      tech="Tech: Postgres, Node.js, JavaScript, Superagent, QUnit"
      title="FrontPage"
    />
  </li>
  <li
    key="2"
  >
    <ProjectItem
      description="ThisÂ was a group project that was created entirely in Vanilla JavaScript. The product was a music sampler and sequencer. With no database, we utilized localStorage to save a user's sequences. My primary contribution was creating the Vanilla JS sequencer array loop that would play an audio sample on-time with the BPM."
      tech="Tech: JavaScript, HTML, CSS, QUnit"
      title="Offbeat"
    />
  </li>
</ul>
`;
